### PR TITLE
[ModuleMapGen] Rewrite only if contents updated

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
- https://bugs.swift.org/browse/SR-2613

### CODE REVIEW

File: Tests/PackageLoadingTests/ModuleMapGenerationTests.swift [hunk 1]
```diff
@@ -128,7 +131,47 @@ class ModuleMapGeneration: XCTestCase {
...
+            try generator.generateModuleMap(inDir: dir)
+
+            let moduleMap = dir.appending(components: "module.modulemap")
+            let contents = try localFileSystem.readFileContents(moduleMap)
+
+            XCTAssert(localFileSystem.isFile(moduleMap))
+            // Store modified time.
+            let modifiedTime = try getModifiedTime(file: moduleMap)
+            // Need to sleep for 1 second because time is precisely stored in seconds.
+            sleep(1)
```
Comment (ddunbar) [inline Tests/PackageLoadingTests/ModuleMapGenerationTests.swift]: I'd rather not have a test that sleeps.

The test I was imagining was a higher level test that whatever base situation Vapor is hitting would result always result in a null second build... I suspect it is a Swift target depending on a C target with a generated module map that triggers it.

Comment (aciidb0mb3r) [inline Tests/PackageLoadingTests/ModuleMapGenerationTests.swift]: Even then sleep has to be there right? for eg: https://github.com/apple/swift-package-manager/blob/master/Tests/FunctionalTests/MiscellaneousTests.swift#L249

**Comment by ddunbar** [comment]
LGTM!

**Comment by abertelrud** [comment]
This is great; thanks @aciidb0mb3r !

**Pull request merged**